src/wccp2.cc

```diff
@@ -2137,7 +2134,7 @@ parse_wccp2_service(void *)
     if ((t = ConfigParser::NextToken()) != NULL) {
         if (strncmp(t, "password=", 9) == 0) {
             security_type = WCCP2_MD5_SECURITY;
-            strncpy(wccp_password, t + 9, WCCP2_PASSWORD_LEN);
+            xstrncpy(wccp_password, t + 9, WCCP2_PASSWORD_LEN);
```

Comment:
Please use sizeof() for the size parameter here to prevent string truncation.
```suggestion
            xstrncpy(wccp_password, t + 9, sizeof(wccp_password));
```

Reply:
Done.